src/logger.ts

```diff
@@ -12,41 +12,37 @@ enum LogLevel {
     DEBUG,
 }
 /* eslint-enable */
-type LogLevelStrings = keyof typeof LogLevel;
+
+const LOG_PREFIX = 'RandomWallpaper';
 
 /**
- *
+ * A convenience logger class.
  */
 class Logger {
-    private _prefix: string;
-    private _callingClass: string;
-    private _settings = new Settings();
-
-    /**
-     * Create a new logging helper.
-     *
-     * @param {string} prefix Custom string to prepend
-     * @param {string} callingClass Class this logger writes messages for
-     */
-    constructor(prefix: string, callingClass: string) {
-        this._prefix = prefix;
-        this._callingClass = callingClass;
-    }
+    private static _SETTINGS: Settings|null = null;
 
     /**
      * Helper function to safely log to the console.
      *
-     * @param {LogLevelStrings} level String representation of the selected log level
+     * @param {LogLevel} level the selected log level
      * @param {unknown} message Message to send, ideally an Error() or string
+     * @param {object} sourceInstance Object where the log originates from (i.e., the source context)
      */
-    private _log(level: LogLevelStrings, message: unknown): void {
+    private static _log(level: LogLevel, message: unknown, sourceInstance?: object): void {
+        if (Logger._selectedLogLevel() < level)
+            return;
+
         let errorMessage = String(message);
 
         if (message instanceof Error)
             errorMessage = message.message;
 
+        let sourceName = '';
+        if (sourceInstance)
+            sourceName = ` >> ${sourceInstance.constructor.name}`;
+
         // This logs messages with GLib.LogLevelFlags.LEVEL_MESSAGE
-        log(`${this._prefix} [${level}] >> ${this._callingClass} :: ${errorMessage}`);
+        log(`${LOG_PREFIX} [${LogLevel[level]}]${sourceName} :: ${errorMessage}`);
```

Comment:
The guide page advises on using `console.log()` and alike.

Same for `logError()` a few lines below. (`console.warn()` ?)

https://gjs.guide/extensions/upgrading/gnome-shell-45.html#logging

Reply:
Those do not work on < 45 yet (I just tried it) ... we can move to it eventually when I can test those.

Reply:
Checked right now and it seems to work.